tool/infra-testing-tool/go/cmd/testflow.go

```diff
@@ -0,0 +1,378 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"infratesting/aws"
+	"infratesting/config"
+	"infratesting/logging"
+	"infratesting/server/grpc/server"
+	"infratesting/testing"
+	"math/rand"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	"berty.tech/berty/v2/go/pkg/messengertypes"
+
+	"github.com/spf13/cobra"
+)
+
+const DefaultTimeout = time.Minute * 5
+
+var (
+	timeout time.Duration
+)
+
+func init() {
+	testFlowCmd.PersistentFlags().DurationVarP(&timeout, "timeout", "t", DefaultTimeout, "test timeout")
+}
+
+type Result struct {
+	started time.Time
+	elapsed time.Duration
+}
+
+var testFlowCmd = &cobra.Command{
+	Use: "test",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := context.Background()
+
+		c, err := loadConfig()
+		if err != nil {
+			return logging.LogErr(err)
+		}
+
+		// @TODO: handle error correctly
+		// need better logging system
+		res, err := runTestFlow(ctx, timeout, c)
+		if err != nil {
+			return logging.LogErr(err)
+		}
+
+		logging.Log(
+			fmt.Sprintf("elpased: %dm%ds", int(res.elapsed.Minutes()), int(res.elapsed.Seconds())),
+		)
+
+		if err := updoadLogs(ctx, res, c); err != nil {
+			return logging.LogErr(err)
+		}
+
+		<-ctx.Done()
+		return ctx.Err()
+	},
+}
+
+func updoadLogs(ctx context.Context, res *Result, c config.Config) error {
+	// handle ctrl+c
+	ctx, cancel := contextHandleSignal(ctx, syscall.SIGINT)
+	defer cancel()
+
+	logging.Log("uploading logs... (ctrl+c to cancel)")
+
+	allNodes, err := testing.GetAllEligiblePeers(aws.Ec2TagType, config.GetAllTypes())
+	if err != nil {
+		return logging.LogErr(err)
+	}
+
+	cerr := make([]chan error, len(allNodes))
+	for n := range allNodes {
+		cerr[n] = make(chan error, 1)
+		go func(n int) {
+			name := allNodes[n].Name
+			resp, err := allNodes[n].P.UploadLogs(ctx, &server.UploadLogs_Request{
+				Folder: strconv.FormatInt(res.started.Unix(), 10),
+				Name:   strings.ReplaceAll(allNodes[n].Tags[aws.Ec2TagName], ".", "-"),
+			})
+			if err != nil {
+				cerr[n] <- logging.LogErr(fmt.Errorf("[%s]: %w", name, err))
+				return
+			}
+
+			bucketName, err := aws.GetBucketName()
+			if err != nil {
+				cerr[n] <- logging.LogErr(fmt.Errorf("[%s]: %w", name, err))
+				return
+			}
+			logging.Log(fmt.Sprintf("%v uploaded: %v files to bucket: %s", allNodes[n].Tags[aws.Ec2TagName], resp.UploadCount, bucketName))
+
+			cerr[n] <- nil
+		}(n)
+	}
+
+	for _, c := range cerr {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case err := <-c:
+			if err != nil {
+				logging.LogErr(err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func runTestFlow(ctx context.Context, timeout time.Duration, c config.Config) (*Result, error) {
```

Comment:
I didn't check in detail but wasn't it possible to reuse the `startTest` and `stopTest` functions to avoid duplicating code / be more DRY ?

Something like:
```go
func runTestFlow() {
	startTest()
	handleSignalAndTimeout()
	stopTest()
}
```